### ISSUE TITLE
Undefined variable: share at lib/private/share20/manager.php#996 

### ISSUE THREAD

**Issue by Dennis1993** [nextcloud/server#499]
### Steps to reproduce
1. share a Folder with remote users (user@example.com)
2. accept the shareing Information
3. user the Folder
4. delete the folder

### Expected behaviour
no error

### Actual behaviour
error in log: Undefined variable: share at cloud/lib/private/share20/manager.php#996 

### Server configuration
**Operating system**:
Ubuntu

**Web server:**
Apache 2.4

**Database:**
MySQL 5.6.

**PHP version:**
7.0.8

**Nextcloud version:** (see Nextcloud admin page)
9.0.53 (stable)

**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh install

**Where did you install Nextcloud from:**
nextcloud.com

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
nothing

**List of activated apps:**
<details>
Default apps with calendar
</details>

**The content of config/config.php:**
<details>
Default, no changes

</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
IE11

**Operating system:**
Windows 10

**Comment by MorrisJobke** [comment]
cc @rullzer @schiessle

**Comment by rullzer** [comment]
This is fixed in f17ad188
